src/Microsoft.AspNetCore.Server.Kestrel/WebHostBuilderKestrelExtensions.cs

```diff
@@ -28,6 +29,7 @@ public static IWebHostBuilder UseKestrel(this IWebHostBuilder hostBuilder)
             return hostBuilder.ConfigureServices(services =>
             {
                 services.AddTransient<IConfigureOptions<KestrelServerOptions>, KestrelServerOptionsSetup>();
+                services.AddTransient<IConfigureOptions<KestrelServerOptions>, KestrelServerConfigureOptions>();
```

Comment:
chatted a bit offline with @davidfowl lets use `services.AddTransient<IConfigureOptions<KestrelServerOptions>, ConfigureDefaults<ConfigureDefaultKestrelServerOptions>()` so when the DI issues is fixed, we only have to remove this line, and won't have to update the ConfigureDefaultOptions class